packages/discord.js-utilities/src/lib/PaginatedMessages/PaginatedMessageEmbedFields.ts

```diff
@@ -0,0 +1,135 @@
+import { PaginatedMessage } from './PaginatedMessage';
+
+/**
+ * This is a utility of {@link PaginatedMessage}, except it exclusively paginates the fields of an embed.
+ * You must either use this class directly or extend it.
+ *
+ * It differs from PaginatedFieldMessageEmbed as the items here are whole fields, that are added to the embed,
+ * whereas PaginatedFieldMessageEmbed concatenates the items in a single field with a given formatter function.
+ *
+ * @example
+ * ```typescript
+ * import { PaginatedMessageEmbedFields } from '@sapphire/discord.js-utilities';
+ *
+ * new PaginatedMessageEmbedFields()
+ * 	.setTemplate({ title: 'Test pager embed', color: '#006080' })
+ * 	.setItems([
+ * 		{ title: 'Sapphire Framework', value: 'discord.js Framework' },
+ * 		{ title: 'Sapphire Framework 2', value: 'discord.js Framework 2' },
+ * 		{ title: 'Sapphire Framework 3', value: 'discord.js Framework 3' }
+ * 	])
+ * 	.setItemsPerPage(2)
+ * 	.make()
+ * 	.run(message);
+ * ```
+ */
+export default class PaginatedMessageEmbedFields extends PaginatedMessage {
+	private embedTemplate: MessageEmbed = new MessageEmbed();
+	private totalPages = 0;
+	private items: EmbedFieldData[] = [];
+	private itemsPerPage = 10;
+
+	/**
+	 * Set the items to paginate.
+	 * @param items The pages to set
+	 */
+	public setItems(items: EmbedFieldData[]): this {
+		this.items = items;
+		return this;
+	}
+
+	/**
+	 * Sets the amount of items that should be shown per page.
+	 * @param itemsPerPage The number of items
+	 */
+	public setItemsPerPage(itemsPerPage: number): this {
+		this.itemsPerPage = itemsPerPage;
+		return this;
+	}
+
+	/**
+	 * Sets the template to be used to display the embed fields as pages. This template can either be set from a template {@link MessageEmbed} instance or an object with embed options.
+	 * All fields in the given template will be overwritten when calling {@link PaginatedMessageEmbedFields.make}.
+	 *
+	 * @param template MessageEmbed
+	 *
+	 * @example
+	 * ```typescript
+	 * import { PaginatedMessageEmbedFields } from '@sapphire/discord.js-utilities';
+	 * import { MessageEmbed } from 'discord.js';
+	 *
+	 * new PaginatedMessageEmbedFields()
+	 * 	.setTemplate(new MessageEmbed().setColor('#006080').setTitle('Test pager embed'))
+	 * 	.setItems([{ name: 'My field', value: 'The field\'s value' }])
+	 * 	.make()
+	 * 	.run(message);
+	 * ```
+	 *
+	 * @example
+	 * ```typescript
+	 * import { PaginatedMessageEmbedFields } from '@sapphire/discord.js-utilities';
+	 * import { MessageEmbed } from 'discord.js';
+	 *
+	 * new PaginatedMessageEmbedFields()
+	 * 	.setTemplate({ title: 'Test pager embed', color: '#006080' })
+	 * 	.setItems([{ name: 'My field', value: 'The field\'s value' }])
+	 * 	.make()
+	 * 	.run(message);
+	 * ```
+	 */
+	public setTemplate(template: MessageEmbed | MessageEmbedOptions): this {
+		this.embedTemplate = template instanceof MessageEmbed ? template : new MessageEmbed(template);
```

Comment:
```suggestion
		this.embedTemplate = this.resolveTemplate(template);
```

Also support an embed callback. 

```ts
private resolveTemplate(template: MessageEmbed | MessageEmbedOptions | ((embed: MessageEmbed) => MessageEmbed)) {
	if (template instanceof MessageEmbed) {
		return template;
	} else if (isFunction(template)) {
		return template(new MessageEmbed());
	}

	return new MessageEmbed(template);
}
```

This way it matches the possibilities for `addPageEmbed`. `isFunction` is from `@sapphire/utilities`
Also add an example for this of course.

Reply:
Will do. Do i also add this to PaginatedFieldMessageEmbed.setTemplate in another commit?

Reply:
Might as well yeah actually